### PR TITLE
retrospective experiments for sarix models

### PR DESCRIPTION
**Pull request by elray1**
varying p, theta pooling strategy, and covariate inclusion

### CODE REVIEW

File: code/sarix_model/configs/sarix_p2_4rt_thetanone_sigmanone.py
Comment (elray1) [inline code/sarix_model/configs/sarix_p2_4rt_thetanone_sigmanone.py:1]: Becky's advice in previous PR for creating a setup based on dataclass with a helper function to instantiate class objects based on passed arguments came too late for this round of updates 😢

File: code/sarix_model/sarix_model.py [hunk 1]
```diff
@@ -46,7 +47,7 @@ def get_sarix_preds(model_config, run_config):
     df['xmas_spike'] = np.maximum(3 - np.abs(df['delta_xmas']), 0)
     
     xy_colnames = ["inc_trans_cs"] + model_config.x
-    batched_xy = df[xy_colnames].values.reshape(len(df['location'].unique()), -1, 2)
+    batched_xy = df[xy_colnames].values.reshape(len(df['location'].unique()), -1, len(xy_colnames))
```
Comment (elray1) [inline code/sarix_model/sarix_model.py:50]: fixing a bug that surfaced when `model_config.x = []`; last dimension should just be 1 in that case.

File: retrospective-hub/model-output/UMass-sarix_p2_4rt_thetanone_sigmanone/2023-10-14-UMass-sarix_p2_4rt_thetanone_sigmanone.csv
Comment (elray1) [inline retrospective-hub/model-output/UMass-sarix_p2_4rt_thetanone_sigmanone/2023-10-14-UMass-sarix_p2_4rt_thetanone_sigmanone.csv:1]: everything from here on down is not really reviewable: it's model output files created from retrospective fits.

File: code/eval/score_sarix_retrospective.R
Comment (elray1) [inline code/eval/score_sarix_retrospective.R:1]: one-off/hasty evaluation script.  look at hubEvals being used in the wild!

File: code/sarix_model/retrospective-experiments/sarix_experiments.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+def run_command(command):
+    """Run system command"""
+    os.system(command)
+
+
+output_root = '../../retrospective-hub/model-output'
+
+commands = [f'python sarix_model.py --ref_date {ref_date} --output_root {output_root} --model_name {model_name}' \
+                for (model_name, ref_date) in product(model_names, ref_dates)]
```
Comment (bsweger) [inline code/sarix_model/retrospective-experiments/sarix_experiments.py]: `subprocess` is the recommended python module for running system commands (it's more secure, is more consistent across platforms, and is more flexible to work with in terms of handling command output and errors).

`subprocess.run` accepts a command list where each component of the command is an individual item. So if you refactor your list of commands along the lines of the suggestion below, you could swap in subprocess without too much effort (and could explore [its parameters](https://docs.python.org/3/library/subprocess.html#subprocess.run) should you ever need to reroute the output or add error handling)

```suggestion
def run_command(command):
    """Run system command"""
    subprocess.run(command)


output_root = '../../retrospective-hub/model-output'

commands = [[
    "python sarix_model.py",
    "--ref-date", ref_date,
    "--output_root", output_root,
    "--model_name", model_name]
    for (model_name, ref_date) in product(model_names, ref_dates)]
```

**Review by bsweger — CHANGES_REQUESTED** [review]
Looks good, but am requesting a change from `os.system` to `subprocess.run` for security and usability reasons (the former is the reason for the change request).

Obviously there's no immediate security threat as you're running experiments, but it's a small change and is a good habit to get into.

**Review by bsweger — APPROVED** [review]

**Pull request merged**